v3/newrelic/sqlparse/sqlparse_test.go

```diff
@@ -192,3 +192,91 @@ func TestExtractTable(t *testing.T) {
 		}
 	}
 }
+
+func TestSqlTokenizerNextWord(t *testing.T) {
+	for input, words := range map[string][]string{
+		"SELECT * FROM table":            {"SELECT", "*", "FROM", "table"},
+		"SELECT * FROM [ table ]":        {"SELECT", "*", "FROM", "[", "table", "]"},
+		"SELECT a,b FROM table":          {"SELECT", "a", ",", "b", "FROM", "table"},
+		"SELECT * FROM (SELECT COUNT())": {"SELECT", "*", "FROM", "(", "SELECT", "COUNT", "(", ")", ")"},
+		"UPDATE rollback{foo}":           {"UPDATE", "rollback", "{", "foo", "}"},
+	} {
+		tokenizer := sqlTokenizer{input}
+
+		for _, word := range words {
+			if next := tokenizer.nextWord(); word != next {
+				t.Errorf("query '%s': expected %s, got %s", input, word, next)
+			}
+		}
+
+		if next := tokenizer.nextWord(); next != "" {
+			t.Errorf("query '%s': expected empty word, got %s", input, next)
+		}
+	}
+}
+
+func TestSqlTokenizerNextToken(t *testing.T) {
+	for input, tokens := range map[string][]string{
+		"SELECT * FROM [ table ]":        {"SELECT", "*", "FROM", "[ table ]"},
+		"SELECT a,b FROM table":          {"SELECT", "a", "b", "FROM", "table"},
+		"SELECT * FROM (SELECT COUNT())": {"SELECT", "*", "FROM", "(SELECT", "COUNT", "))"},
+		"UPDATE rollback{foo}":           {"UPDATE", "rollback", "foo}"},
+	} {
+		tokenizer := sqlTokenizer{input}
+
+		for _, token := range tokens {
+			if next := tokenizer.nextToken(); token != next {
+				t.Errorf("query '%s': expected %s, got %s", input, token, next)
+			}
+		}
+
+		if next := tokenizer.nextToken(); next != "" {
+			t.Errorf("query '%s': expected empty token , got %s", input, next)
+		}
+	}
+}
+
+func TestRemoveAllComments(t *testing.T) {
+	for input, expected := range map[string]string{
+		"query -- comment":                                  "query  ",
+		"query -- comment\nother":                           "query  other",
+		"query /* comment */":                               "query  ",
+		"query /* comment */ other":                         "query  other",
+		"query /* comment */ other /* other comment */ end": "query  other  end",
```

Comment:
I think it would be good to have a test for leading and trailing comments in this set. Perhaps just add a comment at the beginning and the end of this test case?

```suggestion
		"/* leading */ query /* comment */ other /* other comment */ end": "query  other  end /* trailing */",
```

Reply:
I added this test case. I had to add spaces at the end of the result string.